docs/permissions/getting-started.md

```diff
@@ -44,118 +30,102 @@ Asset requests initiated by the browser will not include a token in the `Authori
-   import { PermissionPolicy } from '@backstage/plugin-permission-node';
-   /* highlight-add-start */
-   import {
-     PermissionPolicy,
-     PolicyQuery,
-   } from '@backstage/plugin-permission-node';
-   /* highlight-add-end */
-   import { Router } from 'express';
-   import { PluginEnvironment } from '../types';
-
-   class TestPermissionPolicy implements PermissionPolicy {
-     /* highlight-remove-next-line */
-     async handle(): Promise<PolicyDecision> {
-     /* highlight-add-start */
-     async handle(request: PolicyQuery): Promise<PolicyDecision> {
-       if (request.permission.name === 'catalog.entity.delete') {
-         return {
-           result: AuthorizeResult.DENY,
-         };
-       }
-     /* highlight-add-end */
-
-       return { result: AuthorizeResult.ALLOW };
-     }
-   }
-   ```
-
-3. Now that you’ve made this change, you should find that the unregister entity menu option on the catalog entity page is disabled.
+```yaml title="app-config.yaml"
+permission:
+  enabled: true
+```
+
+2. Its now all wired up and working, great! But perhaps we don't want to simply allow everything, lets try and create our own policy, to do this you can create a new folder in `packages/backend/src` called `permissions` and create a new file called `policy.ts`, in that file we can add the following to create a policy that denies deleting entities from the catalog:
+
+```ts title="packages/backend/src/permissions/policy.ts"
+import {
+  AuthorizeResult,
+  PolicyDecision,
+} from '@backstage/plugin-permission-common';
+import {
+  PermissionPolicy,
+  PolicyQuery,
+} from '@backstage/plugin-permission-node';
+
+export class DenyCatalogDeletePolicy implements PermissionPolicy {
+  async handle(request: PolicyQuery): Promise<PolicyDecision> {
+    if (request.permission.name === 'catalog.entity.delete') {
+      return {
+        result: AuthorizeResult.DENY,
+      };
+    }
+
+    return { result: AuthorizeResult.ALLOW };
+  }
+}
+```
+
+3. We then need to use the permissions backend policy extension point to register our policy, to do this we can add the following to `packages/backend/src/index.ts`:
+
+```ts title="packages/backend/src/index.ts"
+import { createBackend } from '@backstage/backend-defaults';
+import { DenyCatalogDeletePolicy } from './permissions/policy';
+const backend = createBackend();
+// ...
+backend.add(import('@backstage/plugin-permission-backend/alpha'));
+/* highlight-remove-next-line */
+backend.add(
+  import('@backstage/plugin-permission-backend-module-allow-all-policy'),
+);
+/* highlight-add-next-line */
+backend.add(
+  createBackendModule({
+    pluginId: 'permission',
+    moduleId: 'deny-catalog-delete-policy',
+    register(reg) {
+      reg.registerInit({
+        deps: { policy: policyExtensionPoint },
+        async init({ policy }) {
+          policy.setPolicy(new DenyCatalogDeletePolicy());
+        },
+      });
+    },
+  }),
+);
+// ...
+backend.start();
```

Comment:
I am assuming this `/* highlight-add-next-line */` will not work as I want it to here :sweat_smile: or will it?

Reply:
Fixed in [18f5652](https://github.com/backstage/backstage/pull/24229/commits/18f565273a1eb0d30e1b6fc0735ad5df985e2016)